src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -1290,6 +1430,24 @@ static void XMLCALL fmi3_parse_element_data(void* c, const XML_Char *s, int len)
         }
 }
 
+/* Allocate fmi3_xml_parser_context_t with a given xmlType */
+static fmi3_xml_parser_context_t* fmi3_xml_allocate_parser_context(
```

Comment:
I don't think there is any other way to do this really. It is a bit of a hack, but I do see value in actually having the xmlType as const in the structure, we want to make sure it does not change.